### ISSUE TITLE
FATAL EXCEPTION: java.lang.RuntimeException: An error occurred while executing doInBackground()

### ISSUE THREAD

**Issue by berlin4apk** [emschu/snmp-cockpit#3]
FATAL EXCEPTION: pool-1-thread-7
Process: org.emschu.snmp.cockpit, PID: 19868
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:399)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.ClassCastException: org.emschu.snmp.cockpit.y.n.c.o cannot be cast to org.emschu.snmp.cockpit.y.m
	at org.emschu.snmp.cockpit.a0.i.b(TabTaskHelper.java:1)
	at org.emschu.snmp.cockpit.a0.d.b(Unknown Source:0)
	at org.emschu.snmp.cockpit.a0.d.e(SourceFile:3)
	at org.emschu.snmp.cockpit.a0.d.doInBackground(SourceFile:1)
	at android.os.AsyncTask$3.call(AsyncTask.java:378)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more

**Comment by berlin4apk** [comment]
My guess is, it's something in there:

This class was deprecated in API level 30.
Use the standard java.util.concurrent or Kotlin concurrency utilities instead.

AsyncTask was intended to enable proper and easy use of the UI thread. However, the most common use case was for integrating into UI, and that would cause Context leaks, missed callbacks, or crashes on configuration changes. It also has inconsistent behavior on different versions of the platform, swallows exceptions from doInBackground, and does not provide much utility over using Executors directly

https://developer.android.com/reference/android/os/AsyncTask

**Comment by emschu** [comment]
@berlin4apk Thank you very much for reporting this issue. I will have a look at it in the next days or weeks. If you obtain more information about reproducing such errors, feel free to let me know.

**Comment by berlin4apk** [comment]
I will try to make a screen cast
Or have you a good filter for logcat?

**Comment by emschu** [comment]
Dear @berlin4apk,

thank you for your patience so far. I've started to convert the codebase to be compatible with SDK level 30 and migrating the `AsyncTask`s (and a lot of other stuff). But I have very limited testing possibilities and I try to make up for the last years in Android development, which is pretty time-consuming for me (if you don't work with the Android SDK on a daily basis)...

Because of this and a lot of changes in the Android SDK (+ testing libraries such as Espresso), I personally would prefer to rewrite the whole app - if there is time. Probably in september/october/winter this year there is time for this for me. So long I have pushed my recent code migration changes to the branch `feature/api-level-30`, which contains several fixes and improvements, but it is not considered stable so far....

I keep this issue open, until there is a solution.
